lib/jekyll/filters.rb

```diff
@@ -371,7 +371,7 @@ def item_property(item, property)
     # return numeric values as numbers for proper sorting
     def parse_sort_input(property)
       number_like = %r!\A\s*-?(?:\d+\.?\d*|\.\d+)\s*\Z!
-      return property.to_f if property =~ number_like
+      return property.to_f if property&.match?(number_like)
```

Comment:
`property` here can be any Ruby primitive: Array / Hash / String / Integer.., etc
Therefore, we really can't replace `=~` (which is actually defined for [every object](https://ruby-doc.org/core-2.4.0/Object.html#method-i-3D~)) with a method (`match?`) defined only for String and Regexp.

The best solution here would be to disable RuboCop for just this method.